test/t8_cmesh/t8_gtest_attribute_gloidx_array.cxx

```diff
@@ -0,0 +1,135 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <t8_cmesh.h>
+#include <t8_cmesh/t8_cmesh_types.h>
+
+/* Test if multiple attributes are partitioned correctly. */
+
+#define T8_ATTRIBUTE_TEST_MAX_NUM_ENTRIES 1000
+
+class cmesh_attribute_gloidx_array: public testing::TestWithParam<int> {
+ protected:
+  void
+  SetUp () override
+  {
+    num_entries = GetParam ();
+
+    /* Build a cmesh with one QUAD tree and one TRIANGLE tree. */
+    t8_cmesh_init (&cmesh);
+    t8_cmesh_set_dimension (cmesh, 2);
+    t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_QUAD);
+    t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_TRIANGLE);
+
+    /* Allocate space for entries. */
+    entries = T8_ALLOC (t8_gloidx_t, num_entries);
+
+    /* Fill with 0, 1, 2, 3, 4 ... */
+    for (t8_locidx_t ientry = 0; ientry < num_entries; ++ientry) {
+      entries[ientry] = ientry;
+    }
+
+    t8_debugf ("Calling set_attribute with count %i\n", num_entries);
+    /* Set the array as attribute twice. Once with data_persist and once without. */
+    /* Attribute at tree 0, data_persist = 0 */
+    t8_gloidx_t tree_with_attribute = 0;
+    int data_persists = 0;
+    t8_cmesh_set_attribute_gloidx_array (cmesh, tree_with_attribute, t8_get_package_id (), T8_CMESH_NEXT_POSSIBLE_KEY,
+                                         entries, num_entries, data_persists);
+
+    /* Attribute at tree 1, data_persist = 1 */
+    tree_with_attribute = 1;
+    data_persists = 1;
```

Comment:
you can use ```data_persists``` as an additional parameter for the test using ```Combine(generator0, generator1, ...)```. That way, we can simplify the tests down to a single test (the two tests are the same (or I missed a very small difference), which avoids code-duplication.

Reply:
Please check what i did.
i combined the tests into 1.
I still think it good to have 2 trees in the cmesh to cover multiple trees.
Both trees store the attribute but with different data_persist.
we then run the test twice, once for each tree.

Reply:
Yes, this solution is also in the spirit of my remark.